### PR TITLE
Initial PFT Graph Impl

### PR DESCRIPTION
**Pull request by jshaughn**
Closes https://github.com/kiali/openshift-servicemesh-plugin/issues/100

This PR was originally targeted directly for OSSMC but given our more recent approach to native plugins it will now be part of the the main kiali repo.  This makes it more easily updated and copied down to the native plugins along with most of the other frontend code, in bulk.

The PFT graph will be hidden behind an undocumented feature flag, `spec.kiali_feature_flags.ui_defaults.graph.impl`. If if set to 'pf' it will be the graph.  If set to 'both' then both graph impls will be available.

At some point the PFT graph may be removed, or perhaps it will become the default.  Only time will tell.

### CODE REVIEW

**Review by leandroberetta — COMMENTED** [review]
@jshaughn thanks for PR and the huge work on this. 

I tested locally and looks and works good. 

I made some minor comments/questions. I think code looks good, but after this merge we need to aim to remove the files in the components folder (the pf files). 

I think it was a good move to mantain the Cytoscape graph and let the user to decide which one to use as I think the Cytoscape one will be here for a long time.

File: frontend/src/components/JaegerIntegration/JaegerResults/TraceDetails.tsx
Comment (jshaughn) [reply frontend/src/components/JaegerIntegration/JaegerResults/TraceDetails.tsx:82]: It's a bad variable name, it was bad before this PR and it is still bad :-)   It should be something like `graphSelector` or `focusSelector`.  I'll change it...

File: frontend/src/pages/Graph/GraphToolbar/GraphFind.tsx
Comment (jshaughn) [reply frontend/src/pages/Graph/GraphToolbar/GraphFind.tsx:20]: This changed because while creating GraphFindPF, I found a bug in the existing GraphFind.  I fixed it in both impls.

File: frontend/src/pages/Graph/GraphToolbar/GraphFindPF.tsx
Comment (jshaughn) [reply frontend/src/pages/Graph/GraphToolbar/GraphFindPF.tsx:10]: `CyNode` and `CyEdge` are a bunch of string constants for names of data attributes that are stored on Nodes and Edges.  Both Cy and PF allow you to add app-specific "Data" to your elements, basically as an `any` type.  Sometimes we need to refer to those fields by name, and to reduce the chance of typos I have all of these constants.   Unfortunately the current names include "Cy" and defined in CytoscapeGraphUtil.ts.  I guess now is a good time to change it because you are right, it's not great to see "Cy" in the PF impl.  I was just avoiding it because I couldn't think of a good, short name to replace `CyNode` and `CyEdge`.   I'll change it to `NodeAttr` and `EdgeAttr`, longer but ok I guess.

Note that we use the same data attributes in both impls.  In fact the PF graph uses the same JSON that we send from the backend.  That JSON is configured well for Cytoscape, but I just adapt it to be loaded into PF.

**Comment by jshaughn** [comment]
@leandroberetta I pushed changes in response to your comments.

> ... but after this merge we need to aim to remove the files in the components folder (the pf files).

I'm not sure about this comment, can you explain more what you mean?  I don't think we will plan to remove the PF graph impl in the near future.  It will be used as the default for the native plugins, and just hidden behind the feature flag in Kiali.

**Comment by leandroberetta** [comment]
> @leandroberetta I pushed changes in response to your comments.
> 
> > ... but after this merge we need to aim to remove the files in the components folder (the pf files).
> 
> I'm not sure about this comment, can you explain more what you mean? I don't think we will plan to remove the PF graph impl in the near future. It will be used as the default for the native plugins, and just hidden behind the feature flag in Kiali.

I was referring to the PF files that were copied to make customizations.

**Comment by jshaughn** [comment]
> I was referring to the PF files that were copied to make customizations.

Ah, OK, right.  I don't know how yet, but if we can get away from that approach it would be great.

**Review by leandroberetta — DISMISSED** [review]
Thanks for addressing the comments. 

LGTM!

**Comment by jshaughn** [comment]
@leandroberetta , I resolved all of the conflicts after rebasing against Fernando's "export default" refactor.  You will need to re-approve if things still look good.

**Review by jmazzitelli — APPROVED** [review]

**Pull request merged**
